### PR TITLE
[jit] test builtin types

### PR DESCRIPTION
**Pull request by lewurm**
not sure about the `arch_pointer_size` stuff in `Makefile.am.in`.  is there a way to get my hands on `SIZEOF_VOID_P` in the Makefile?

### CODE REVIEW

**Comment by alexrp** [comment]
You can access `SIZEOF_VOID_P` from makefiles if you do an `AC_SUBST(SIZEOF_VOID_P)` somewhere in configure.ac.

File: mono/mini/mini-native-types.c [hunk 1]
```diff
@@ -361,6 +364,7 @@ mono_class_is_magic_int (MonoClass *klass)
 	static MonoClass *magic_nint_class;
 	static MonoClass *magic_nuint_class;
 
+
 	if (klass == magic_nint_class)
```
Comment (vargaz) [inline mono/mini/mini-native-types.c]: Extra line.

File: mono/mini/builtin-types.cs [hunk 2]
```diff
@@ -0,0 +1,1371 @@
...
+		return 0;
+	}
+
+	public static int Main (String[] args) {
+		return TestDriver.RunTests (typeof (BuiltinTests), args);
+	}
+}
+
+
+// !!! WARNING - GENERATED CODE - DO NOT EDIT !!!
```
Comment (vargaz) [inline mono/mini/builtin-types.cs]: Another reason we don't run this test suite is that these definitions don't correspond a 100% to the definitions in the product.

Comment (lewurm) [inline mono/mini/builtin-types.cs]: I'm not sure what is worse: (1) not testing at all or (2) testing something potentially slightly off.

What prevents us to move this code from XI to mono nowadays? Except for the magic check, nothing else seems to be specific to the product.

Comment (vargaz) [inline mono/mini/builtin-types.cs]: There is nowhere to put them, they are part of the XI api.

Comment (alexrp) [inline mono/mini/builtin-types.cs]: Couldn't we include them as a source-only library along the lines of Mono.Options?

Comment (akoeplinger) [inline mono/mini/builtin-types.cs]: I'd say (2) is better :)

The problem is that these are generated by a T4 template so it relies on VSMac to generate the .cs source which makes it a bit cumbersome to add in Mono. You can take a look at  https://github.com/xamarin/xamarin-macios/tree/master/src/NativeTypes and at least refresh the copy.

**Comment by lewurm** [comment]
- added `SIZEOF_VOID_P` to `configure.ac` so it's available in Makefiles.
- imported a more recent version for `System.{nint,nuint,nfloat}` from `xamarin-macios` as suggested by @akoeplinger.
- disabled some broken tests. some seem new'ish, some look like they never worked.
- disabled it on `fullaot`.  we can't directly compile e.g. `.ctor`. if we want to run the test in `fullaot`, we have to blacklist certain methods somehow.
- `linux/armv7` and `win/x64` failures are unrelated.

good to squash&merge?

**Comment by vargaz** [comment]
Looks ok.

**Comment by vargaz** [comment]
Should delete the version in mono-extensions after this is merged.

**Pull request merged**